### PR TITLE
Remove undocumented `room_alias` key from `/createRoom` response

### LINKED ISSUE #10321: Remove undocumented `room_alias` key from `/createRoom` response
we should either spec this or remove it.

Presumably it exists to save clients the bother of having to construct their own fully-qualified aliases, but given we don't do the same for other endpoints which create aliases, I'm not convinced it is worthwhile.

I'd therefore be in favour of removing it, but it's worth noting that sytest currently uses it in a non-trivial number of places

### PR DESCRIPTION
**Pull request by thezaidbintariq**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

Fixes #10321

### CODE REVIEW

**Comment by clokep** [comment]
@thezaidbintariq Could you please [sign-off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off) on your contribution?

File: changelog.d/10321.removal [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Removed  if room_alias: 
+     result["room_alias"] = room_alias.to_string() 
+from synapse/synapse/handlers/room.py
+Contributed by @thezaidbintariq.
```
Comment (clokep) [inline changelog.d/10321.removal:4]:
```suggestion
Remove undocumented `room_alias` key from `/createRoom` response. Contributed by @thezaidbintariq.
```

Comment (thezaidbintariq) [reply changelog.d/10321.removal:4]: Signed-off-by: Zaid Bin Tariq ztariq.bscs19seecs@seecs.edu.pk

**Comment by thezaidbintariq** [comment]
Signed-off-by: Zaid Bin Tariq ztariq.bscs19seecs@seecs.edu.pk

**Comment by thezaidbintariq** [comment]
@clokep did I sign-off correctly?

File: changelog.d/10321.removal [hunk 2]
```diff
@@ -0,0 +1 @@
+Remove undocumented `room_alias` key from `/createRoom` response. Contributed by @thezaidbintariq.
```
Comment (anoadragon453) [inline changelog.d/10321.removal:1]: The number in the changelog's filename will need to match the number of your PR (14884). Could you rename the changelog file to match that number?

**Review by anoadragon453 — COMMENTED** [review]
@thezaidbintariq Your sign-off is correct :+1:

It would be good for someone to shout in the [Matrix Client Developers Room](https://matrix.to/#/#matrix-client-developers:matrix.org) about this change before including it in a release of Synapse - just to ensure that no client desperately relies on this (unspecced) field.

File: changelog.d/10321.removal (hunk 2 above)
Comment (thezaidbintariq) [reply changelog.d/10321.removal]: @anoadragon453 Done, could you please verify if I did it correctly?

**Comment by thezaidbintariq** [comment]
@anoadragon453 can you merge the PR?

**Comment by clokep** [comment]
> It would be good for someone to shout in the [Matrix Client Developers Room](https://matrix.to/#/#matrix-client-developers:matrix.org) about this change before including it in a release of Synapse - just to ensure that no client desperately relies on this (unspecced) field.

This was done at https://matrix.to/#/!XXSJTvRPInupfUgQVb:matrix.org/$xxmBTJfTuP2icgn0X2xfDsygfUR3JbsFYQAMBE1eMPs?via=pixie.town&via=matrix.org&via=element.io

**Comment by anoadragon453** [comment]
> > It would be good for someone to shout in the [Matrix Client Developers Room](https://matrix.to/#/#matrix-client-developers:matrix.org) about this change before including it in a release of Synapse - just to ensure that no client desperately relies on this (unspecced) field.
> 
> This was done at [matrix.to/#/!XXSJTvRPInupfUgQVb:matrix.org/$xxmBTJfTuP2icgn0X2xfDsygfUR3JbsFYQAMBE1eMPs?via=pixie.town&via=matrix.org&via=element.io](https://matrix.to/#/!XXSJTvRPInupfUgQVb:matrix.org/$xxmBTJfTuP2icgn0X2xfDsygfUR3JbsFYQAMBE1eMPs?via=pixie.town&via=matrix.org&via=element.io)

I bumped the question as nobody had responded yet.

**Comment by DMRobertson** [comment]
> This is failing due to complement requiring the `room_alias` field, it lokos like this was removed from [matrix-org/sytest#1216](https://github.com/matrix-org/sytest/pull/1216), but never ported to complement, see https://github.com/matrix-org/complement/blob/4418d2c05009cb0754403a9a8565b8b701633a5b/tests/csapi/apidoc_room_create_test.go#L39

I have opened https://github.com/matrix-org/complement/pull/616 for this.

**Comment by DMRobertson** [comment]
(I am happy to take this over, to get this across the line)

**Comment by DMRobertson** [comment]
Complement should now pass now that https://github.com/matrix-org/complement/pull/616 has landed

**Comment by DMRobertson** [comment]
Hmm. On reflection it looks like there's a bit more internal plumbing to be done here than we'd anticipated. I'm going to have a go at this in #15093.

**Comment by DMRobertson** [comment]
> Hmm. On reflection it looks like there's a bit more internal plumbing to be done here than we'd anticipated. I'm going to have a go at this in #15093.

This has just merged. Thank you for your efforts looking into this.

**Pull request closed without merging**